Source/inv.cpp

```diff
@@ -2111,7 +2084,7 @@ void CleanupItems(int ii)
 
 	int i = 0;
 	while (i < numitems) {
-		if (itemactive[i] == ii) {
+		if (itemactive[i] == FindGetItem(item.IDidx, item._iCreateInfo, item._iSeed)) {

```

Comment:
What's the intent of this change?

Reply:
We don't have the index for the item anymore, so we have to fetch it based on the item.

Reply:
this could potentially find a duplicate, maybe better to also pass the index?

at least we should only do the search ones (outside the loop)

Reply:
> this could potentially find a duplicate

I'm using the same function used in the duplication logic though. You sure this could get a duplicated item?

> at least we should only do the search ones (outside the loop)

Indeed, that was a big oversight. I'll change it.

Reply:
What duplication logic ;)

Reply:
> What duplication logic ;)

The item duplication checks use this method's conditions to define what a duplicate item means.

Reply:
The code was a bit faulty and could end up marking legit items as duplicates (especially uniques), since we fixed the duplication bug the faulty anti-duplication code has been removed.